pymatgen/symmetry/analyzer.py

```diff
@@ -1697,3 +1700,543 @@ def __str__(self):
 
     def __repr__(self):
         return self.__str__()
+
+
+class MagneticSpacegroupAnalyzer:
+    """
+    Takes a pymatgen.core.structure.Structure object, symprec and angle_tolarance.
+
+    Determines magnetic spacegroup operations and magnetic spacegroup. 
+    Depends on pymatgen.symmetry.analyzer.SpacegroupAnalyzer, which interfaces
+    with spglib.
+
+    Args:
+        structure (Structure/IStructure): Structure to find symmetry
+        symprec (float): Tolerance for symmetry finding of SpacegroupAnalyzer. 
+            Defaults to 0.01, which is fairly strict and works well for properly 
+            refined structures with atoms in the proper symmetry coordinates. For
+            structures with slight deviations from their proper atomic
+            positions (e.g., structures relaxed with electronic structure
+            codes), a looser tolerance of 0.1 (the value used in Materials
+            Project) is often needed.
+        angle_tolerance (float): Angle tolerance for symmetry finding.
+        pos_tolerance (float): Tolerance for fractional coordinates in magnetic 
+            structures. Defaults to 0.0015.
+        magmom_tolerance (float): Tolerance for the magnetic moment. 
+            Defaults to 0.3, which is about the uncertainty one would get in an
+            ab initio calculation.
+    """
+
+    def __init__(self, structure, symprec=0.01, angle_tolerance=5, pos_tolerance=0.0015, magmom_tolerance=0.3):
+        # TODO: structure code such that the circular dependencies btw pymatgen.transformations 
+        #       and pymatgen.symmetry are resolved
+      
+        from pymatgen.transformations.standard_transformations import ApplyMagSymmOpTransformation
+        from pymatgen.alchemy.materials import TransformedStructure
+      
+        self._structure = structure
+        self._symprec = symprec
+        self._angle_tol = angle_tolerance
+        self._pos_tol = pos_tolerance
+        self._magmom_tol = magmom_tolerance
+        self._domains = None
+      
+        self._spacegroupAnalyzer = SpacegroupAnalyzer(self._structure, 
+                                              symprec = self._symprec, 
+                                              angle_tolerance = self._angle_tol
+                                              )
+      
+        if np.any(np.abs(self._structure._lattice.matrix - self._spacegroupAnalyzer._cell[0]) > 1e-12): 
+            exit("Error in the definitions of the unit cell.")
```

Comment:
`exit()` might make some users sad since it would quit their Python session, instead an Exception should be raised. For example, `raise VallueError("Error in the definitions of the unit cell.")`. 

For approximation comparisons, `np.all_close()` is a nice short-hand for the `np.any(np.abs(...) > ...)` construction.